test/lib/NonLinearDutchDecayLib.t.sol

```diff
@@ -500,10 +507,12 @@ contract NonlinearDutchDecayLibTest is Test, GasSnapshot {
 
         NonlinearDutchDecay memory curve = CurveBuilder.multiPointCurve(blocks, decayAmounts);
 
-        uint256 currentBlock = decayStartBlock + blockDelta;
         vm.roll(currentBlock);
-        uint256 decayed = NonlinearDutchDecayLib.decay(curve, startAmount, decayStartBlock, 0, startAmount);
-
-        assertEq(decayed, startAmount - decayAmountFuzz, "Should be fully decayed for block delta beyond uint16.max");
+        uint256 decayed = NonlinearDutchDecayLib.decay(curve, startAmount, decayStartBlock, minAmount, maxAmount);
+        assertEq(
+            decayed,
+            Math.max(startAmount - decayAmountFuzz, minAmount),
```

Comment:
if we are restricting to the case where an overflow happens, shouldn't decayed always equal minAmount and nothing else?

Reply:
You can set a `minAmount` that's lower than the last decay of a curve. In the extreme case someone could even put 0

Reply:
ah yes because you also specify the relativeAmounts